src/ucp/proto/proto_common.c

```diff
@@ -397,8 +407,11 @@ void ucp_proto_common_calc_perf(const ucp_proto_common_init_params_t *params,
     iface_attr = ucp_proto_common_get_iface_attr(&params->super,
                                                  perf_params->lane0);
     frag_size  = ucp_proto_get_iface_attr_field(iface_attr,
-                                                params->max_frag_offs, SIZE_MAX) -
-                 params->hdr_size;
+                                                params->max_frag_offs, SIZE_MAX);
+    if (!(params->flags & UCP_PROTO_COMMON_INIT_FLAG_RESPONSE)) {
+        /* if the data returns as a response, no need to subtract header size */
```

Comment:
can some proto send data with another header?

Reply:
do you mean whether the response data may have a header as well, in addition to request?
currently no such protocol is expected so probably no need to add such support for now

Reply:
@yosefe maybe document this assumption in the code.